src/main/java/refinedstorage/tile/TileDestructor.java

```diff
@@ -71,9 +95,29 @@ public int getEnergyUsage() {
     @Override
     public void updateNode() {
         if (ticks % upgrades.getSpeed(BASE_SPEED, 4) == 0) {
-            if (type == IType.ITEMS) {
-                BlockPos front = pos.offset(getDirection());
 
+            BlockPos front = pos.offset(getDirection());
+
+            if (pickupItem && type == IType.ITEMS) {
+                List<Entity> droppedItem = new ArrayList<>();
+
+                Chunk chunk = worldObj.getChunkFromBlockCoords(front);
+                chunk.getEntitiesWithinAABBForEntity(null, new AxisAlignedBB(front), droppedItem, null);
+
+                if (droppedItem.size() > 0) {
```

Comment:
The for each following the statement will not loop when size is Zero. So no it isn't needed